### ISSUE TITLE
"!ERROR = !SELECT: no such aggregate 'sql_div'" for time operations

### ISSUE THREAD

**Issue by AKheli** [MonetDB/MonetDB#7249]
I want to aggregate the average and the weighted average of the same column and have them side-by-side in the results. However, this is giving me the following exception: `ERROR = !SELECT: no such aggregate 'sql_div'` and now `"!syntax error, unexpected IDENT in: "select"`. I reckon this is due to the way I am extracting the seconds from the data.  

I tried to adapt the query from the [documentation](https://www.monetdb.org/documentation-Jan2022/user-guide/sql-functions/date-time-functions/) but I had no luck.

```
QUERY = SELECT
           id_station,
           avg(pH) as mean,
           sum(
              pH *
              (60 - extract("second"(timetz '2019-06-01 17:00','%Y/%m/%d %H:%M') - interval '1' hour))
           ) / (60 * 61 / 2) as weighted_mean
        FROM
           datapoints
        WHERE
           -- Last 60 periods
           time > timestamptz '2019-06-01 17:00'
        GROUP BY
           id_station;
ERROR = !syntax error, unexpected IDENT in: "select
        !   id_station,
        !   avg(ph) as mean,
        !   sum(
        !      ph *
        !      (60 - extract"
CODE  = 42000
```

**Comment by PedroTadim** [comment]
I have to say there are a lot of inconsistencies here. There are two group by clauses, a where clause after group by, "extract" is a special sql syntax so it should be used as extract(second from ....), you have a timetz value with a timestamp string?

I think you want something like this:

`SELECT`
`	id_station,`
`	avg(pH) as mean,`
`	sum(`
`		pH *`
`		(60 - extract(second from (timetz '17:00' - interval '1' hour)))`
`	) / (60 * 61 / 2) as weighted_mean`
`	FROM`
`		datapoints`
`	WHERE`
`		"time" > timestamptz '2019-06-01 17:00'`
`	GROUP BY`
`		id_station;`

**Comment by AKheli** [comment]
Hello @PedroTadim. I added the first GROUP BY for testing reasons (to run the query without the WHERE statement). I removed it from the issue question. The problem in the issue title was produced by the first query with one GROUP BY statement. 

What I want with the extract statement is to extract the _seconds_ from the datetime format _'2019-06-01 17:00'_ and not just the time _'17:00'_ as in `extract(second from (timetz '17:00' - interval '1' hour))`. I replaced the _timez_ with a _timestamptz_ in the query and the query worked as wanted. 

```
SELECT
id_station,
avg(pH) as mean,
sum(
pH *
(60 - extract(second from (timestamptz '2019-06-01 17:00' - interval '1' hour)))
) / (60 * 61 / 2) as weighted_mean
FROM
datapoints
WHERE
"time" > timestamptz '2019-06-01 17:00'
GROUP BY
id_station;
```
Thanks!
